### PR TITLE
A18 assignments Q1, Q2, Q3 and Q4.

### PR DESCRIPTION
**Pull request by tmmrajesh**

### CODE REVIEW

File: Ops/Expr/Evaluate.cs [hunk 1]
```diff
@@ -2,24 +2,45 @@
...
+         "tan" => Math.Tan (D2R (args[0])),
+         "log" => Math.Log (args[0]),
+         "exp" => Math.Exp (args[0]),
+         "asin" => R2D (Math.Asin (args[0])),
+         "acos" => R2D (Math.Acos (args[0])),
+         "atan" => R2D (Math.Atan (args[0])),
+         "atan2" => R2D (Math.Atan2 (args[0], args[1])),
+         "random" => Random.Shared.Next (),
+         "round" => args.Length == 1 ? Math.Round (args[0]) : Math.Round (args[0], (int)args[1]),
+         _ => throw new NotImplementedException ()
```
Comment (tarydon) [inline Ops/Expr/Evaluate.cs:38]: Fine for now, but we need an extensible system once people start writing their own functions in Pascal. Besides, even this list of possible function should be obtained from some "Runtime Library DLL"...

File: Parser/Parser.cs [hunk 2]
```diff
@@ -67,6 +72,17 @@ NExpr Expression ()
       return expr;
    }
 
+   NExpr[] ArgList () {
+      if (Match (CLOSE)) return Array.Empty<NExpr> ();
+      List<NExpr> args = new ();
+      for (; ; ) {
+         args.Add (Expression ());
+         if (!Match (COMMA)) break;
+      }
```
Comment (tarydon) [inline Parser/Parser.cs]: `do args.Add (Expression ()); while (!Match (COMMA));`

File: Parser/Token.cs [hunk 3]
```diff
@@ -41,6 +41,28 @@ public enum E {
       Console.ForegroundColor = ConsoleColor.Yellow;
       Console.WriteLine ($"At line {Line}, column {Column} of {Source.FileName}: {Text}");
       Console.ResetColor ();
+      // Print the title.
+      var title = $"File: {Source.FileName}";
+      Console.WriteLine (title);
+      // Underline the title.
+      const int gutter = 5;
+      Console.WriteLine ("┬".PadLeft (gutter, '─').PadRight (title.Length, '─'));
```
Comment (tarydon) [inline Parser/Token.cs:49]: Very nice!

File: Ops/Expr/Evaluate.cs (hunk 1 above)
Comment (tmmrajesh) [reply Ops/Expr/Evaluate.cs:38]: OK.  Hopefully part of another assignment :). Thanks.

**Pull request closed without merging**